### ISSUE TITLE
Fales positive on nullable member field inside a future

### ISSUE THREAD

**Issue by wbadam** [uber/NullAway#1159]
NullAway reports that a field that's resolved inside a future is nullable despite a protective condition.

It seems this effects **member fields of any class**, but not local fields such as function arguments.

```java
    @Nullable String nullableMemberField;

    public int aFunction(@Nullable String nullableFunctionArgument) {
        if (nullableFunctionArgument != null) {
            var future = CompletableFuture.completedFuture("foo").thenApply(foo -> {
                return nullableFunctionArgument.length(); // <-- no report
            });
            return future.join();
        }
        if (nullableMemberField != null) {
            var future = CompletableFuture.completedFuture("foo").thenApply(foo -> {
                return nullableMemberField.length(); // <-- warning: [NullAway] dereferenced expression nullableMemberField is @Nullable
            });
            return future.join();
        }
    }
```

A reproducible simple project is here: https://github.com/wbadam/nullaway-java-sample/blob/adam.w-pattern-matching/lib/src/main/java/net/sridharan/sample/Library.java

**Comment by msridhar** [comment]
This is because in general, it's possible that before that lambda passed to `thenApply` actually runs, `nullableMemberField` gets nulled out again, which could lead to an NPE.  As it so happens, for cases like these, we just added better support for `@MonotonicNonNull` annotations in NullAway 0.12.4.  See the docs:

https://github.com/uber/NullAway/wiki/Supported-Annotations#initialization

If `nullableMemberField` never gets explicitly assigned a `@Nullable` value, then I think writing `@MonotonicNonNull String nullableMemberField` should work here and get rid of the warning.  Does that work for your use case?

**Comment by wbadam** [comment]
Oh, thank you for the explanation, i somehow assumed NullAway isn't clever enough to understand lambdas run in the future. It makes sense, will close this then.
